src/FixedRankMatrices.jl

```diff
@@ -0,0 +1,177 @@
+using LinearAlgebra: diag, Diagonal, svd, SVD, rank, dot
+
+@doc doc"""
+    FixedRankMatrices{M,N,K} <: Manifold
+
+The manifold of $m\times n$ real-valued matrices of fixed rank $k$, i.e.
+````math
+\mathcal M = \{ x \in \mathbb R^{m\times n} : \operatorname{rank}(x) = k \}
+````
+
+# Representation with 3 matrix factors
+A point $x\in\mathcal M$ can be stored using orthonormal matrices
+$U\in\mathbb R^{m\times k}$, $V\in\mathbb R^{n,k}$ as well as the $k$ singular
+values of $x = USV^\mathrm{T}$. To be precise, $U$ and $V$ are from the manifolds
+[`Stiefel`](@ref)`(n,k)` and [`Stiefel`](@ref)`(m,k)`, respectively; see
+[`SVDMPoint`](@ref) for details
+
+The tangent space $T_x\mathcal M$ at a point $x\in\mathcal M$ with $x=USV^\mathrm{T}$
+is given by
+````math
+T_x\mathcal M = \bigl\{ UMV^\mathrm{T} + U_xV^\mathrm{T} + UV_x^\mathrm{T} : 
+    M \in \mathbb R^{k\times k},
+    U_x \in \mathbb R^{m\times k},
+    V_x \in \mathbb R^{n\times k}
+    \text{ s.t. }
+    U_x^\mathrm{T}U = 0_k,
+    V_x^\mathrm{T}V = 0_k
+\bigr\},
+````
+where $0_k$ is the $k\times k$ zero matrix. See [`UMVTVector`](@ref) for details.
+
+This representation follows
+> Bart Vandereycken: "Low-rank matrix completion by Riemannian Optimization,
+> SIAM Journal on Optiomoization, 23(2), pp. 1214–1236, 2013.
+> doi: [10.1137/110845768](https://doi.org/10.1137/110845768),
+> arXiv: [1209.3834](https://arxiv.org/abs/1209.3834).
+"""
+struct FixedRankMatrices{M,N,K,T} <: Manifold end
+FixedRankMatrices(m::Int, n::Int, k::Int, T::Type = Type{Real}) = FixedRankMatrices{m,n,k,T}()
+
+@doc doc"""
+    SVDMPoint <: MPoint
+
+A point on a certain manifold, where the data is stored in a svd like fashion,
+i.e. in the form $USV^\mathrm{T}$, where this structre stores $U$, $S$ and
+$V^\mathrm{T}$. The storage might also be shortened to just $k$ singular values
+and accordingly shortened $U$ (columns) and $V^\mathrm{T}$ (rows)
+
+# Constructors
+* `SVDMPoint(A)` for a matrix `A`, stores its svd factors
+* `SVDMPoint(S)` for an `SVD` object, stores its svd factors
+* `SVDMPoint(U,S,Vt)` for the svd factors to initialize the `SVDMPoint``
+* `SVDMPoint(A,k)` for a matrix `A`, stores its svd factors shortened to the
+  best rank $k$ approximation
+* `SVDMPoint(S,k)` for an `SVD` object, stores its svd factors shortened to the
+  best rank $k$ approximation
+* `SVDMPoint(U,S,Vt,k)` for the svd factors to initialize the `SVDMPoint`,
+  stores its svd factors shortened to the best rank $k$ approximation
+"""
+struct SVDMPoint{T} <: MPoint
```

Comment:
To make it more compatible with `StaticArrays` and AD, I recommend more abstract typing here, something like
```julia
struct SVDMPoint{TU<:AbstractMatrix, TS<:AbstractVector, TVt<:AbstractMatrix} <: MPoint
    U::TU
    S::TS
    Vt::TVt
end
```